File: stage/src/app/model/cours.ts
import { CategorieCours } from "./catgorie";
import { Enseignant } from "./enseignant";

export class Cours{
    id!:number;
    name!:string;
    file!:string;
    enseignant!:Enseignant;
    categoriecours!:CategorieCours;
    constructor(id:number,name:string,file:string){
        this.id=id;
        this.name=name;
        this.file=file;
    }
}

File: UniversityProject-master/UniversityProject-master/src/main/java/com/example/private_university/Entity/Cours.java
package com.example.private_university.Entity;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity @NoArgsConstructor @AllArgsConstructor @Data
public class Cours {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id ;
    private String name ;
    private String file ;
    @ManyToOne
    private Enseignant enseignant;
    @ManyToOne
    private CategorieCours categorieCours ;
}
